src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
   }
 
   // return the canvas ref for operations like toDataURL
-  getCanvas = () => {
+  getCanvas = (): HTMLCanvasElement => {
     return this._canvas
   }
 
   // return a trimmed copy of the canvas
-  getTrimmedCanvas = () => {
+  getTrimmedCanvas = (): HTMLCanvasElement => {
     // copy the canvas
     const copy = document.createElement('canvas')
     copy.width = this._canvas.width
     copy.height = this._canvas.height
-    copy.getContext('2d').drawImage(this._canvas, 0, 0)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    copy.getContext('2d')!.drawImage(this._canvas, 0, 0)
     // then trim it
     return trimCanvas(copy)
   }
 
   // return the internal SignaturePad reference
-  getSignaturePad = () => {
+  getSignaturePad = (): SignaturePad => {
     return this._sigPad
   }
 
-  _checkClearOnResize = () => {
+  _checkClearOnResize = (): void => {
     if (!this.props.clearOnResize) {
       return
     }
     this._resizeCanvas()
   }
 
-  _resizeCanvas = () => {
-    const canvasProps = this.props.canvasProps || {}
+  // a few eslint disables of strict-boolean-expressions here because changing
+  // these lines would be breaking -- 0s, '', null etc are falsey
+  _resizeCanvas = (): void => {
+    const canvasProps = this.props.canvasProps ?? {}
     const { width, height } = canvasProps
     // don't resize if the canvas has fixed width and height
-    if (width && height) {
+    if (typeof width !== 'undefined' && typeof height !== 'undefined') {
       return
     }
 
     const canvas = this._canvas
     /* When zoomed out to less than 100%, for some very strange reason,
       some browsers report devicePixelRatio as less than 1
       and only part of the canvas is cleared then. */
-    const ratio = Math.max(window.devicePixelRatio || 1, 1)
+    const ratio = Math.max(window.devicePixelRatio ?? 1, 1)
 
-    if (!width) {
+    if (typeof width === 'undefined') {
       canvas.width = canvas.offsetWidth * ratio
     }
-    if (!height) {
+    if (typeof height === 'undefined') {
       canvas.height = canvas.offsetHeight * ratio
     }
-    canvas.getContext('2d').scale(ratio, ratio)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    canvas.getContext('2d')!.scale(ratio, ratio)
     this.clear()
   }
 
-  render () {
+  render: Component['render'] = () => {
     const { canvasProps } = this.props
-    return <canvas ref={(ref) => { this._canvas = ref }} {...canvasProps} />
+    return <canvas ref={this._setRef} {...canvasProps} />
   }
 
   // all wrapper functions below render
   //
-  on = () => {
+  on: SignaturePad['on'] = () => {
     window.addEventListener('resize', this._checkClearOnResize)
     return this._sigPad.on()
   }
 
-  off = () => {
+  off: SignaturePad['off'] = () => {
     window.removeEventListener('resize', this._checkClearOnResize)
     return this._sigPad.off()
   }
 
-  clear = () => {
+  clear: SignaturePad['clear'] = () => {
```

Comment:
I think most of the following methods can be simplified to e.g:
```
clear = this._sigPad.clear
```

Reply:
`_sigPad` doesn't get set until `componentDidMount`, so that doesn't work. I wish it were easier, but I don't think there's a way of doing it